src/Nethermind/Nethermind.Blockchain.Test/Producers/BuildBlocksOnlyWhenNotProcessingTests.cs

```diff
@@ -35,8 +35,7 @@ public async Task should_trigger_block_production_when_queue_empties()
         context.BlockProcessingQueue.IsEmpty.Returns(false);
         Task<Block?> buildTask = context.MainBlockProductionTrigger.BuildBlock();
 
-        await Task.Delay(BuildBlocksOnlyWhenNotProcessing.ChainNotYetProcessedMillisecondsDelay * 2);
-        buildTask.IsCanceled.Should().BeFalse();
+        Assert.That(() => buildTask.IsCanceled, Is.False.After(BuildBlocksOnlyWhenNotProcessing.ChainNotYetProcessedMillisecondsDelay * 2, 10));
```

Comment:
It might be cool to have a method that receives timeout only and be default checks 5 times or so

Reply:
You can achieve that with current API by diving the timeout with the number of checks, not sure if worth confusing people.